app/controllers/api/v3/analytics/user_analytics_controller.rb

```diff
@@ -30,32 +43,132 @@ def respond_to_html_or_json(stats)
     end
   end
 
-  def first_of_current_month
-    Date.current.at_beginning_of_month
+  def prepare_daily_stats
```

Comment:
I thought about it, but it felt overkill since `UserAnalytics` is a relatively cordoned-off aspect of the app. We have a lot of "abstractions" now. Query objects, database models, materialized views – it felt too much to add another layer.

Happy to stick a `Presenter` class in if you feel the controller is too heavy.

Reply:
Yeah the controller does seem quite heavy. I'm still leaning towards putting all this stuff into a separate presenter that can handle these methods, as well as any data massaging logic that's still in the view.